javascript/lifecycle.js

```diff
@@ -13,7 +14,7 @@ import { camelize } from './utils'
 // - element - the element that triggered the reflex (not necessarily the Stimulus controller's element)
 //
 const invokeLifecycleMethod = (stage, element, reflexId) => {
-  if (!element || !element.reflexData[reflexId]) return
+  if (!element || !element.reflexData || !element.reflexData[reflexId]) return
```

Comment:
We should probably warn the user if the lifecycle method couldn't be executed. If we just return here the user is probably wondering why his lifecycle method wasn't executed - because there is no indication or error messages which would suggest that the execution failed.

Reply:
As I write this, I'm still in the grokking phase but if we go down this path, we just have to make sure that we're only emitting warnings if debugging is true. That's actually a bit tricky because the functions in `lifecycle.js` don't currently have access to the `debugging` flag. This is because `invokeLifecycleMethod` is never invoked directly by the SR codebase! It's *only* invoked by the event listeners in `lifecycle.js` that drive the library's behaviour.

That means we'd have to figure out a way to pass the `debugging` flag through. Our options for that are:
- global (`window`) variable
- attach `debugging` variable to `BODY`
- add `data-reflex-debugging` attribute to body
- add `META` tag to HEAD
- move contents of `lifecycle.js` back into `stimulus_reflex.js` so that `debugger` is in file scope

I like/hate some of those more than others. @hopsoft will need to render an opinion. 🔢

Reply:
Alright, I just implement a solution for this (even though it wasn't one on your list 🙈). 
Everywhere you want to access the current debugging value you can now `impor  Debugg from './debug.js'`. With this you can access the value of it in every file you want with `Debug.enabled` or just `Debug.value`, always returns a boolean.

I you think we need another solution let me know!